### ISSUE TITLE
Adding example for the latest Image captioning model by Baidu

### ISSUE THREAD

**Issue by cognitronz** [fchollet/keras#265]
Hi,

I have been going through the latest the LSTM model for the text generation example from Andrej Kaparthy.  It's pretty cool and works pretty well.  Now, I am wondering if the LSTM model by Junhua Mao (http://arxiv.org/abs/1412.6632) can be supported easily.  The reason i am asking this is the duplication of an embedding layer before the RNN (figure2).  I guess one can use the RepeatVector, but I havent seen enough examples to be sure.  So some clarification will be helpful.

Thanks!  Keras rocks!

**Comment by fchollet** [comment]
So, how would you implement it in Keras? I'd be happy to check out you code and give you feedback.

For the record, RepeatVector does not split the data flow into multiple outputs. The output of RepeatVector is still a single tensor (where the input vector gets repeated n times). However we are working on a Fork system that would allow for splitting a data flow into multiple outputs.

**Comment by cognitronz** [comment]
I wasnt saying I know how to do it, actually.  I was wondering if you
already thought about it.  I will try to play with it.  Without a fork, it
would be hard.  Let me know when the fork feature is complete.  Thanks!

On Mon, Jun 22, 2015 at 7:36 PM, François Chollet <notifications@github.com>
wrote:

> So, how would you implement it in Keras? I'd be happy to check out you
> code and give you feedback.
>
> For the record, RepeatVector does not split the data flow into multiple
> outputs. The output of RepeatVector is still a single tensor (where the
> input vector gets repeated n times). However we are working on a Fork
> system that would allow for splitting a data flow into multiple outputs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fchollet/keras/issues/265#issuecomment-114333114>.
>



-- 
Thomas Zheng

858-740-4398

**Comment by fchollet** [comment]
This is now possible thanks to the introduction of the Graph model: http://keras.io/models/#graph

Will you build it?

**Comment by cognitronz** [comment]
Thanks for the heads up.  I will definitely try to build it.  I hope it
will work.

On Sat, Jul 4, 2015 at 3:44 PM, François Chollet <notifications@github.com>
wrote:

> This is now possible thanks to the introduction of the Graph model:
> http://keras.io/models/#graph
>
> Will you build it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fchollet/keras/issues/265#issuecomment-118560939>.
>



-- 
Thomas Zheng

858-740-4398

**Comment by kracwarlock** [comment]
Do you have any example ready where you show how to use the graph as a recurrent model for image captioning?
The part where I have difficulty understanding is: At training time you can use the ground truth word for the previous time-step as the next input but at test time you need to sample from the softmax at each time-step. So we would need to create LSTM() with length 1 and then keep on passing the parameters to next time-step? Or is there a better way to do this?
